src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -69,4 +60,27 @@ private SwiftDescriptions() {
     output.enableObjcInterop = Optional.of(true);
   }
 
+  static boolean hasSwiftSource(ImmutableSortedSet<SourceWithFlags> srcs) {
+    return Iterables.any(srcs, new Predicate<SourceWithFlags>() {
+      @Override
+      public boolean apply(SourceWithFlags input) {
+        SourcePath sourcePath = input.getSourcePath();
+        return (sourcePath instanceof PathSourcePath) &&
```

Comment:
Good points about the project generation. `halide_library` used to implement `findDepsForTargetFromConstructorArgs` in a similar way but then we removed it because it was adding complexity for nothing.

The dependency on the placeholder swift library can be added in `createBuildRule`, `findDepsForTargetFromConstructorArgs` just makes it easier to add, but the additional complexity down the line is not worth it.

Can you change `AppleLibraryDescription.createBuildRule` and `AppleBinaryDescription.createBuildRule` to do that and remove the changes to `findDepsForTargetFromConstructorArgs`?